pkg/sidecar/sidecar.go

```diff
@@ -36,105 +35,60 @@ type Config struct {
 	SvidKeyFileName          string `hcl:"svidKeyFileName"`
 	SvidBundleFileName       string `hcl:"svidBundleFileName"`
 	RenewSignal              string `hcl:"renewSignal"`
-	Timeout                  string `hcl:"timeout"`
 	ReloadExternalProcess    func() error
+	Log                      logger.Logger
 }
 
 // Sidecar is the component that consumes the Workload API and renews certs
 // implements the interface Sidecar
 type Sidecar struct {
-	config            *Config
-	processRunning    int32
-	process           *os.Process
-	workloadAPIClient workload.X509Client
-	certReadyChan     chan struct{}
+	config         *Config
+	processRunning int32
+	process        *os.Process
+	certReadyChan  chan struct{}
 }
 
 const (
-	// default timeout Duration for the workloadAPI client when the defaultTimeout
-	// is not configured in the .conf file
-	defaultTimeout = 5 * time.Second
-	delayMin       = time.Second
-	delayMax       = time.Minute
-
 	certsFileMode = os.FileMode(0644)
 	keyFileMode   = os.FileMode(0600)
 )
 
 // NewSidecar creates a new SPIFFE sidecar
-func NewSidecar(config *Config) (*Sidecar, error) {
-	timeout, err := getTimeout(config)
-	if err != nil {
-		return nil, err
+func NewSidecar(config *Config) *Sidecar {
+	if config.Log == nil {
+		config.Log = logger.Null
 	}
-
 	return &Sidecar{
-		config:            config,
-		workloadAPIClient: newWorkloadAPIClient(config.AgentAddress, timeout),
-		certReadyChan:     make(chan struct{}, 1),
-	}, nil
+		config:        config,
+		certReadyChan: make(chan struct{}),
```

Comment:
Changing this to an unbuffered channel breaks the current semantics provided by `CertReadyChan`. The only callers in this repository are tests, which reach in here to replace this channel (which is a bit gross), but who knows if other folks are using this package. Is there a reason for this change that I'm not seeing?

Reply:
I don't remember for sure, but i think its not necessary to have the buffer so thats why i removed it. I put it back because we should probably have a notice for that type of change.

Reply:
If it isn't buffered then a race is introduced between callers of CertReadyChan and updateCertificates. If consumers of  CertReadyChan are not selecting on the channel before updateCertificates tries to insert into certReadyChan then the insert will be dropped (because of the `default`) and consumers won't be notified until the next cert update (even though they should have been notified for the original update). It for sure needs to be buffered as currently designed.

Reply:
yes, thats a good point.